scripts/install_hotel.sh

```diff
@@ -0,0 +1,146 @@
+    echo "$token_result"
+    log "found github token in macOS keyring"
+  else
+    if [ -f "$hotel_secrets_path/$account_name" ]; then
+      cat "$hotel_secrets_path/$account_name"
+      log "found github token in $hotel_secrets_path/$account_name"
+    else
+      log "no github token found: $hotel_secrets_path/$account_name does not exist"
+      return 1
+    fi
+  fi
+}
+
+is_github_token_valid() {
+  scopes=$(curl -sLI -u "_:$1" "https://api.github.com/user" |
+    grep '^x-oauth-scopes: ' |
+    sed 's/^x-oauth-scopes: //')
+  if [ "$scopes" = "" ]; then
+    log "token not valid"
+    return 1
+  fi
+  if ! echo "$scopes" | grep "repo" >/dev/null 2>&1; then
+    log "token does not have the 'repo' scope"
+    return 1
+  fi
+}
+# }}}
+
+get_and_store_github_key() {
+  set +e
+  existing_token=$(retrieve_github_token)
+  find_token_result=$?
+  set -e
+  if [ $find_token_result -eq 0 ] && is_github_token_valid "$existing_token"; then
+    # if token exists and is valid we can successfully exit this function
+    echo "$existing_token"
+    return 0
+  fi
+  log ""
+  log "We need a github key to download hotel, and for hotel to use to pull git repos"
+  log "It will be stored in the system keychain"
+  log "You can get this from:"
+  log "    https://github.com/settings/tokens/new?scopes=repo "
+  log ""
+  # no token found, ask user
+  read -s -r -p "Github token: " github_token
+  if ! is_github_token_valid "$github_token"; then
+    log "=> provided token not valid"
+    exit 1
+  fi
+  log "=> provided token okay"
+  store_github_token "$github_token"
+  echo "$github_token"
+}
+
+download_latest_hotel() {
+  github_token="$1"
+  if [ -z "$github_token" ]; then
+    log "github_token missing"
+    exit 1
+  fi
+
+  # we can't get the specific release we want without a json parsing tool, so we get all
+  # download links and download until we find the one matching the system's arch and os
+  release_asset_urls=$(curl -sL -u "_:$github_token" https://api.github.com/repos/cultureamp/hotel/releases/latest |
+    grep '"url": ".*/releases/assets/.*"' |
+    cut -d\" -f4)
+
+  for url in $release_asset_urls; do
+    # the only way to get a release's file name is to download it and write-out the filename
+    downloaded_file=$(curl -sL "$url" \
+      -u "_:$github_token" \
+      --remote-header-name --remote-name \
+      --write-out "%{filename_effective}" \
+      --header "Accept: application/octet-stream")
+
+    if [ "$downloaded_file" = "$hotel_tarball_name" ]; then
+      tar -xzf "$downloaded_file"
+      return
+    else
+      rm "$downloaded_file"
+    fi
+  done
+}
+
+install_hotel() {
+  log "=> installing hotel, this will ask for a sudo password"
+  INITIAL_DIR="$PWD"
+  TMPDIR=$(mktemp -d)
+  cd "$TMPDIR"
+  download_latest_hotel "$1"
+  mkdir -p "$hotel_bin_path"
+  mv hotel "$hotel_bin_path"
+  cd "$INITIAL_DIR"
+  rm -rf "$TMPDIR"
```

Comment:
```suggestion
  sudo mv hotel "$hotel_bin_path"
  cd "$INITIAL_DIR"
  rm -rf "$TMPDIR"
  sudo mkdir /usr/local/bin
```
The above suggestion is a possible solution to get the installer to run on the first try. 
In a fresh VM with macos I found there was no `/usr/local/bin` dir. However I'm confused as to why the `mv` command requires `sudo`.

Reply:
Good catch, have fixed inside hotel, can you review this one https://github.com/cultureamp/hotel/pull/39